implement notes.md for competencies (#23)
Cyber4All/competency-ecosystem-docs: docs/competency-api/notes.md

---
layout: default
title: Notes
nav_order: 8
parent: Competency API
---
# Notes

## Update Notes

### Description
This route is used to update the Notes section of a competency with any additional information that the author would like to include.

#### Patch
```http
https://api.competency-constructor.clark.center/competencies/:competencyId/notes
```

### Headers
```json
{
    Authorization: "Bearer Example.bearer.token"
}
```

### Request

| Name | Required | Type | Description |
|:----:|:-----:|:----:|-----|

> | notes | false | string | Any notes related to the competency |

Reviewer: ```suggestion
| details | true | string | Any notes related to the competency |
```
`notes` is actually `details` and it is required

<details closed markdown="block">
  <summary>
    Example
  </summary>

### Example Http request body
```json
{
    body: {

>         notes: "This competency is meant for students with a Cyber Security track only."

Reviewer: ```suggestion
        details: "This competency is meant for students with a Cyber Security track only."
```

    }
}
```

### Example Curl request
```bash
curl -X PATCH \
  -H "Content-Type: application/json" \
  -H "Authorization": "Bearer Example.bearer.token" \
  -d '{

>         notes: "This competency is meant for students with a Cyber Security track only."

Reviewer: ```suggestion
        details: "This competency is meant for students with a Cyber Security track only."
```

    }' \

>   -L "https://api.competency-constructor.clark.center/competencies/6112745b84804cf5833aa94c/behavior"

Reviewer 2: ```suggestion
  -L "https://api.competency-constructor.clark.center/competencies/6112745b84804cf5833aa94c/notes"
```

```
</details>

### Response

| Name | Description |
|:----:|----|
| 200 | Notes updated successfully |

> | 400 | Competency is deprecated, rejected, or published and cannot be updated. Or body is incorrectly formatted. |

Reviewer 3: I left this comment on Andy's and Paige's PRs but Can it be made more obvious that this is a list of 400 errors?
